src/components/app.tsx

```diff
@@ -88,7 +88,7 @@ export const authAndConnect = (stores: IStores, onQAClear?: (result: boolean, er
     .then(async ({appMode: newAppMode, authenticatedUser, classInfo, problemId, unitCode}) => {
       // authentication can trigger appMode change (e.g. preview => demo)
       if (newAppMode && (newAppMode !== appMode)) {
-        setAppMode(stores, newAppMode);
+        stores.appMode = newAppMode;
```

Comment:
Yeah you're right. I thought for a while that MobX would automatically wrap observables with actions. And actually it kind of does, but it doesn't batch up the changes. Also the current approach is to only warning when a property is changed if it is actually observed by something. So interestingly this property never complained because apparently it isn't observed before it is set here. In any case I made a setter for it to be safe.